### ISSUE TITLE
Issue Getting Stuff

### ISSUE THREAD

**Issue by acitinnovation** [Azure-Samples/active-directory-dotnet-graphapi-web#15]
Hello,

here is my problem :
I have create the Azure website, and publish everything on it. Now when I go on the site and I sign in with my account it work great. I can see the message in the top right corner "Hello, my@email.com !"

If I go in the User profile section everything work great and I see my info. The issue is when I want to see another section. Let say "UserList"

If I click on the UserList link I see the page but the table is empty (And my AAD is full of users) and I see the message "You have to sign-in to see Users. Click here to sign-in."

If I click on the link the re-sign-in then I have an infinite loop that is starting.

Can you guide me on what could I have done wrong ?

Thanks !

**Comment by mgmiles** [comment]
I  am having exactly the same issue. Does anyone what the problem is?

Thanks.

**Comment by wycoconut** [comment]
I have the same problem as well.  Really appreciate it if someone can help us out.

**Comment by SharePointRadi** [comment]
If you look at the exception it will tell you that you have insufficient permissions. The actual exception handling is not fantastic in the sample. 

To get it working (I just did) go to you Azure portal, your directory, open the application configuration, make sure you give it delegated permissions:

![delegated](https://cloud.githubusercontent.com/assets/5693618/7332451/6908b320-eb48-11e4-89e9-aa95cf10c401.png)

**Comment by pandrew1** [comment]
I'm getting this same issue even when running the sample on my local machine. The example Azure AD directory works fine. I've added all 6 of the permissions in the screenshot from SharePointRadi and after that I cleared cookies. But I'm still getting the error. Any other ideas for how to debug this?

**Comment by shawntabrizi** [comment]
Hi All,
I ran into this same issue, and figured out that the problem with with my Reply URL configuration in the sample. I had followed the instructions before, setting my Reply URL to localhost, but then when moving to my own Azure Website, I didnt update it, which seems to cause this infinite loop.

Make sure your <add key="ida:PostLogoutRedirectUri" value="<REPLY URL>" /> is correct in the web.config!
